iohyve

```diff
@@ -258,9 +258,25 @@ __get_zfs_pcidev_conf() {
 
 # List Guests
 __list() {
+	local flags="$1"
 	local pools="$(zfs list -H | grep iohyve | cut -d '/' -f 1 | uniq )"
 	(
-        printf "%s^%s^%s^%s^%s\n" "Guest" "VMM?" "Running" "rcboot?" "Description"
+	
+	# Print common header
+        printf "%s^%s^%s^%s" "Guest" "VMM?" "Running" "rcboot?"
+	
+	# Print headers for -v flag (tap, con)
+	if [ -n "$(echo "$flags" | grep 'v')" ]; then
+		printf "^%s^%s" "Tap" "Con"
+	fi
+	
+	# Print headers for -d flag (description)
+	if [ -n "$(echo "$flags" | grep 'd')" ]; then
+		printf "^%s" "Description"
+	fi
+	
+	printf "\n" # Print new line for first guest.
```

Comment:
Using this method, the flags can be used al-la-carte without needing to write static blocks of code for the desired output.